### ISSUE TITLE
Node 3180: public IPs are not working

### ISSUE THREAD

**Issue by coesensbert** [threefoldtech/zos#1898]
We had a temporary vm running for the Vector POC: https://docs.grid.tf/threefold/itenv_threefold_main/src/branch/master/Grid-deployments/Mainnet/poc-vector/main.tf

It has a public config (IPv4&6) but it's not reachable anymore:
185.69.167.221
2a02:1802:5e:0:3099:35ff:fe8c:3d05

But the VM is still 'reachable' over ygg: 300:efd:2293:9398:9eba:3d55:b25a:513a

All contracts are in a 'Created' state:
```
query MyQuery {
  nodeContracts(where: {contractID_eq: "9549"}) {
    id
    createdAt
    contractID
    nodeID
    numberOfPublicIPs
    state
  }
}
query MyQuery2 {
  nodeContracts(where: {contractID_eq: "9550"}) {
    id
    createdAt
    contractID
    nodeID
    numberOfPublicIPs
    state
  }
}
query MyQuery3 {
  nodeContracts(where: {contractID_eq: "9551"}) {
    id
    createdAt
    contractID
    nodeID
    numberOfPublicIPs
    state
  }
}
```

Terraform output:

```
➜  poc-vector git:(master) ✗ terraform show                  
# grid_deployment.d1:
resource "grid_deployment" "d1" {
    id                = "9551"
    ip_range          = "10.164.3.0/24"
    network_name      = "pocvectornetw"
    node              = 3180
    solution_provider = 0

    disks {
        name = "root"
        size = 300
    }

    vms {
        computedip  = "185.69.167.221/24"
        computedip6 = "2a02:1802:5e:0:3099:35ff:fe8c:3d05/64"
        corex       = false
        cpu         = 4
        description = "Threefold Vector poc"
        env_vars    = {
            "SSH_KEY" = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIDYNeJXJV2FNEwuQz6e0jkKeqRbKWwftBKq+sjSTqa2x"
        }
        flist       = "https://hub.grid.tf/tf-official-vms/ubuntu-22.04-lts.flist"
        ip          = "10.164.3.2"
        memory      = 8192
        name        = "vectorpoc"
        planetary   = true
        publicip    = true
        publicip6   = true
        rootfs_size = 0
        ygg_ip      = "300:efd:2293:9398:9eba:3d55:b25a:513a"
        zlogs       = []

        mounts {
            disk_name   = "root"
            mount_point = "/data"
        }
    }
}

# grid_network.net1:
resource "grid_network" "net1" {
    access_wg_config   = <<-EOT
        
        [Interface]
        Address = 100.64.164.2
        PrivateKey = eJFNZZ4f37xga3yv2tXE/ZsBWtOP6i5eP5LbzUK0X2k=
        [Peer]
        PublicKey = Kx8GRNT7jmC0BPnlDI/5br6y+b/z12la72US8WKUd2o=
        AllowedIPs = 10.164.0.0/16, 100.64.0.0/16
        PersistentKeepalive = 25
        Endpoint = 185.69.166.140:2552
        
    EOT
    add_wg_access      = true
    description        = "Threefold vector poc network"
    external_ip        = "10.164.2.0/24"
    external_sk        = "eJFNZZ4f37xga3yv2tXE/ZsBWtOP6i5eP5LbzUK0X2k="
    id                 = "3e7dcf08-eb1b-48c7-a777-86ac001a9da5"
    ip_range           = "10.164.0.0/16"
    name               = "pocvectornetw"
    node_deployment_id = {
        "1"    = 9550
        "3180" = 9549
    }
    nodes              = [
        3180,
    ]
    nodes_ip_range     = {
        "1"    = "10.164.4.0/24"
        "3180" = "10.164.3.0/24"
    }
    public_node_id     = 1
}


Outputs:

node1_vm1_ip = "10.164.3.2"
public_ip = "185.69.167.221/24"
public_ip6 = "2a02:1802:5e:0:3099:35ff:fe8c:3d05/64"
wg_config = <<-EOT
    
    [Interface]
    Address = 100.64.164.2
    PrivateKey = eJFNZZ4f37xga3yv2tXE/ZsBWtOP6i5eP5LbzUK0X2k=
    [Peer]
    PublicKey = Kx8GRNT7jmC0BPnlDI/5br6y+b/z12la72US8WKUd2o=
    AllowedIPs = 10.164.0.0/16, 100.64.0.0/16
    PersistentKeepalive = 25
    Endpoint = 185.69.166.140:2552
    
EOT
ygg_ip = "300:efd:2293:9398:9eba:3d55:b25a:513a"
```

ZOS node logs: https://mon.grid.tf/explore?orgId=1&left=%5B%22now-2d%22,%22now%22,%22Loki%22,%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnetwork%3D%5C%22production%5C%22,node%3D%5C%223180%5C%22%7D%22%7D%5D

**Comment by muhamadazmy** [comment]
Finding so far:
- Host has 2 VMs, the other VM public ip is working fine, this indicate that the node plumbing in general in a healthy state.
- the affected VM also looks good, since i was granted access to it over Ygg, setting the default gw of the VM to use the user private network resource allowed the VM to reach the internet as expected
- @LeeSmet noticed that the tap device on the host is running in `PROMISC` mode. that's the only difference between the public tap devices on the host! 
- I tried to turn the PROMISC off, and tried to ping the vm again over the public ip, i was successful but with a VERY high drop rate (around 80%)
- @delandtj promised to look into it today, it sounds like an IP conflict (another device in the same network is using the same ip) 
- Another important question, what/how the tap device switched to `PROMISC`

**Comment by LeeSmet** [comment]
- After @muhamadazmy removed the "promiscuous mode" from the tap interface on the host, we got _some_ connectivity back, with about 80-90% packet loss.
- In the VM, the default gateway had previously been switched to the private IP. Switching it back to the public IP restored public IP networking
  - It is not clear why we got _some_ replies for pings. The replies were correctly send over the the private IP as per the rules, so there is no reason we saw any replies (packet loss should have been 100%).
  - It is not clear how promiscuous mode affected the interface, if at all. `dmesg` shows that the interface entered promiscuous mode on 15th of november, which is probably when the VM and thus the interface was created, though the public network used to work fine. This would mean there is something else going on.
- It is not possible to SSH from the VM to the host, but it is possible to SSH to neighbouring hosts (after adding a rule for the host network over the private ip, i.e. `ip r add 10.13.0.0/16 via 10.146.3.1`). This makes sense: `ndmz` is connected to the host ns via `npub4`, a macvlan with zos as parent, and macvlans don't allow directo communication between child and parent. Since the host IP is part of the zos bridge, this does not work.
  - In the past, this was possible on some GE nodes. I'm not 100% sure why (it's been a while), but the working theory is that this actually worked because we SSH'ed over the public IP, and the router allowed public traffic to enter private network segmetns (VM pub IP, with dst internal network -> Router, looks up route to dst (internal network -> Host). This theory is currently unconfirmed though.
  - Another theory is that it should be possible to SSH to the host from the VM via the IPv6 (if the host has one) of the private interface: `ndmz` will forward IPv6 traffic on `npub6`, which also a macvlan, but attached to `br-pub` on the host, which in single nic mode is attached via veth pair to the `zos` bridge. Since `br-pub` is not the target, the macvlan should not complain and properly forward data.
